File: latencySlowload/Latency/LatencyProblem.java
package Latency;

public class LatencyProblem {

    public static String processTask() {
        try {
            // Simulasi tugas berat
            Thread.sleep(5000);
            System.out.println("Synchronous Task Completed!");
            return "Task Completed";
        } catch (InterruptedException e) {
            e.printStackTrace();
            return "Task Failed";
        }
    }

    public static void main(String[] args) {
        System.out.println("Starting task...");
        String result = processTask();
        System.out.println("Result: " + result);
    }
}

File: latencySlowload/LatencyPython/LatencyProblem.py
import asyncio


async def process_async_task():
    print("Task started...")
    await asyncio.sleep(5)  # Simulasi tugas berat dengan penundaan 5 detik
    print("Asynchronous Task Completed!")
    return "Task Completed"


async def main():
    print("Starting task...")

    # Menjalankan tugas asinkron untuk menghindari latency
    future = asyncio.create_task(process_async_task())

    # Lakukan tugas lain sementara menunggu future selesai
    print("Doing other work while waiting...")

    result = await future  # Menunggu hasil dari tugas asinkron
    print(f"Result: {result}")


# Menjalankan loop asyncio
asyncio.run(main())
